src/_guides/language/sound-dart.md

```diff
@@ -37,41 +37,46 @@ developer experience.
 Strong mode Dart adds only a few additional rules beyond that for classic
 Dart&mdash;mostly you clarify code where the types are ambiguous or
 incorrect. In fact, most strong mode errors can be fixed by adding type
-annotations to your Lists and Maps. The following code shows valid
-but unsound Dart code. The `fn()` function prints an integer list.
-The `main()` function creates a list of integers and passes it to `fn()`:
+annotations to your Lists and Maps.
+For example, in the following code the `fn()` function prints an integer list,
+and `main()` creates a list and passes it to `fn()`.
+In classic Dart, the analyzer reports no issues and the code runs without errors.
 
 {:.fails-sa}
 {% prettify dart %}
-void fn(List<int> a) {
-  print(a);
-}
+void fn(List<int> a) => print(a);
 
-main() {
+void main() {
   var list = [];
   list.add(1);
   list.add("2");
   fn([!list!]);
 }
 {% endprettify %}
 
-In classic Dart, this code passes analysis with no errors. Once you enable
-strong mode, a warning appears on `list` (highlighted above) in the call to
-`fn(list)`. The warning states **Unsound implicit cast from List&lt;dynamic&gt;
-to List&lt;int&gt;**. The `var list = []` code creates a list of type
-`dynamic` because the static analyzer doesn't have enough information to
-infer a type.
-The `fn` function expects a list of type `int`, causing a mismatch of types.
+If you enable strong mode, a type error is reported on `list` (highlighted
+above) at the call of `fn(list)`:
+
+```nocode
+The argument type 'List' can't be assigned to the parameter type 'List<int>'
+```
+
+The error, reported at runtime and by the analyzer (when
+[implicit casts are disabled](#call-dartanalyzer-with-strong-mode-enabled)),
+highlights an unsound implicit cast from `List<dynamic>` to `List<int>`.
+The `list` variable has static type `List<dynamic>`. This is because the
+initializing declaration `var list = []` doesn't provide the analyzer with
+enough information for it to infer a type more specific than `dynamic`.
```

Comment:
Drive-by comment: At this point, maybe `type argument` is better than `type`, given that we could actually type an expression evaluating to that list as `dynamic`. Similarly, we could say `a list of type `List<int>`, ..` in line 70, in order to avoid some potential confusion among readers.

Reply:
Good points. I've made both changes.